src/transformers/models/maskformer/feature_extraction_maskformer.py

```diff
@@ -288,11 +297,30 @@ def _max_by_axis(self, the_list: List[List[int]]) -> List[int]:
                 maxes[index] = max(maxes[index], item)
         return maxes
 
+    def convert_segmentation_map_to_binary_masks(self, segmentation_map: "np.ndarray", num_labels: int):
+        # get all the labels in the image
+        labels = np.unique(segmentation_map)
+        # remove ignore index
+        labels = labels[labels != self.ignore_index]
```

Comment:
Does this mean that the segmentation maps should already include the `ignore_index` when providing them to the feature extractor?

This is different from SegformerFeatureExtractor for instance, which assumes the segmentation maps can have a value of 0, which it will replace by 255 if `reduce_labels` is set to True.

Reply:
You need to pass the correct `ignore_index` based on the index you want to remove, if it is zero, then you should pass zero. Maybe I missed something, let me know if it makes sense to you

Reply:
Hmm annotations from ADE20k have 0 in them, not 255, right?

Reply:
If all the rest defaults to values of ADE20k, then let's `ignore_index` also default to 0

Reply:
so as far as I know you do that when you have instances, so you do -1 and set your ignore index at 255 (max range of pixel value), the user has to do the shift

Reply:
Hmm this is not consistent with SegFormer, I would either make it consistent or state it clearly in the docs

Reply:
You are right, adding it